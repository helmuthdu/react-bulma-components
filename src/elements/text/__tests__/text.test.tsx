import { render } from '@testing-library/react';
import * as React from 'react';
import { Text } from '..';

describe('Text component', () => {
  it('should render', () => {
    const { asFragment } = render(<Text />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should concat classname in props with classname', () => {
    const { asFragment } = render(<Text className="other-class this-is-a-test" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should use inline styles', () => {
    const { asFragment } = render(<Text style={{ height: 250 }} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should have size 3 and rendered as span', () => {
    const { asFragment } = render(
      <Text textSize={3} as="span">
        Subtitle
      </Text>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
