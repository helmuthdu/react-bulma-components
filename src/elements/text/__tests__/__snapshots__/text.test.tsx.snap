// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component should concat classname in props with classname 1`] = `
<DocumentFragment>
  <p
    class="other-class this-is-a-test"
  />
</DocumentFragment>
`;

exports[`Text component should have size 3 and rendered as span 1`] = `
<DocumentFragment>
  <span
    class="is-size-3"
  >
    Subtitle
  </span>
</DocumentFragment>
`;

exports[`Text component should render 1`] = `
<DocumentFragment>
  <p />
</DocumentFragment>
`;

exports[`Text component should use inline styles 1`] = `
<DocumentFragment>
  <p
    style="height: 250px;"
  />
</DocumentFragment>
`;
